platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py

```diff
@@ -13,12 +13,19 @@
 
     from .platform_thrift_client import ThriftClient
     from .platform_thrift_client import thrift_try
+    from .platform_thrift_client import pltfm_mgr_try
 
     from sonic_platform_base.sfp_base import SfpBase
     from sonic_platform_base.sonic_sfp.sfputilbase import SfpUtilBase
+    from sonic_py_common import device_info
 except ImportError as e:
     raise ImportError (str(e) + "- required module not found")
 
+SFP_TYPE = "SFP"
+QSFP_TYPE = "QSFP"
+QSFP_DD_TYPE = "QSFP_DD"
+
+
 class SfpUtil(SfpUtilBase):
```

Comment:
Please use Sfp refactoring because SFP handling in SONiC has moved to mostly platform independent code. Ref: [SFP refactoring HLD](https://github.com/Azure/SONiC/blob/bf657839e521fb71e407df18e566a3e09c7e6958/doc/sfp-refactor/sfp-refactor.md).

Use [SfpOptoeBase](https://github.com/Azure/sonic-platform-common/blob/master/sonic_platform_base/sonic_xcvr/sfp_optoe_base.py#L10) class instead of SfpUtilBase (which is going to be phased out)

Example Dell platform moved to using SFP refactored code : - [https://github.com/Azure/sonic-buildimage/pull/9016](https://github.com/Azure/sonic-buildimage/pull/9016)

Reply:
@prgeor , thank you for reviewing. Totally make sense. Looks like redesigning into sonic_xcvr approach and testing will take some time. Can we have these changes merged as-is (mainly because of other components) and refactor SFP part in a separate PR?

Reply:
sonic_xcvr is already tested so your platform is good to use sonic_xcvr approach. There is very little code that is platform specific when it comes to supporting the transceivers, so very less platform specific code means less testing and bugs for your platform. please let me know if you face any issue while moving to sonic_xcvr.